File: src/main/java/algorithm/lc/SearchA2DMatrix.java
package algorithm.lc;

public class SearchA2DMatrix {
	public static class Solution {
		public boolean searchMatrix(int[][] matrix, int target) {
			int m = matrix.length;
			if (m == 0) {
				return false;
			}      
			int n = matrix[0].length;
			if (n == 0) {
				return false;
			}
			int r = 0;
			int c = n - 1;
			while (r < m && c >= 0) {
				if (matrix[r][c] == target) {
					return true;
				} else if (matrix[r][c] < target) {
					--c;
				} else {
					++r;
				}
			}
			return false;
		}
	}
}

File: src/main/python/lc/search_a_2d_matrix.py
class Solution:
  # @param matrix, a list of lists of integers
  # @param target, an integer
  # @return a boolean
  def searchMatrix(self, matrix, target):
    m = len(matrix)
    if m == 0:
      return False
    n = len(matrix[0])
    if n == 0:
      return False
    r = 0
    c = n - 1
    while r < m and c >= 0:
      if matrix[r][c] == target:
        return True
      elif matrix[r][c] < target:
        r += 1
      else:
        c -= 1
    return False     
